### DISCUSSION TITLE
Checking if an asset exists

### DISCUSSION THREAD

**Discussion by tasruMLT** [ranaroussi/yfinance#1401] [Q&A]
I am trying to retrieve a list of assets from my broker and evaluate if they meet certain criteria. However, I have noticed that Yahoo Finance does not always have information on all assets. I have tried using the .info method to determine if an asset exists on Yahoo Finance, but this method is not always reliable. I have also checked if the market capitalization exists or is empty, but this is not always an accurate way to check.

Is there a more reliable method for determining if an asset exists on Yahoo Finance without calling the .info method?

**Comment by pineapplentrees** [comment]
you need a comprehensive list of all US stocks, you can filter them buy exchanges (NASDAQ, SP500, Russell etc)

**Reply by ValueRaider** [reply]
#402 might help

**Comment by ValueRaider** [comment]
If you're asking for an alternative call in `yfinance`, then no 100% reliable method because Yahoo can return nothing if you appear to be spamming. Best option is `Ticker.history`, least likely to trigger spam (less server load):

```
try:
  df = yf.Ticker(tkr).history(period="1d", raise_errors=True)
except:
  # Not listed?
```

**Reply by tasruMLT** [reply]
Thanks. I will check once encryption issue is sorted out

**Reply by ValueRaider** [reply]
Encryption doesn't break everything. Price history and `fast_info` works.
